app/js/components/pages/Flow.jsx

```diff
@@ -56,35 +46,44 @@ const Flow = ({ routeParams: { pluginId, flowId } }, { store }) => {
                 <br />
                 <br />
 
-            { flow.outputArtifacts.map(({ name, type }) =>
+            { workflow.outputArtifacts.map(({ name, type }) =>
                 <fieldset
                     className="form-group"
                     key={ `${name}-text-output${counter++}` }
                 >
-                    <label htmlFor="out-{ name }">
+                    <label htmlFor={`out-${name}`}>
                         Output Name: { name }
                     </label>
                     <input
                         type="text-field"
                         className="form-control"
-                        name="out-{ name }"
+                        name={`out-${name}`}
                         placeholder={ type }
                     />
                 </fieldset>
             )}
-
-
-                <button type="submit" className="btn btn-primary">Go!</button>
             </form>
+            <button
+                className="btn btn-primary"
+                onClick={onClickSubmit}
+            >
+                Go!
+            </button>
```

Comment:
I would flip the order of these in the UI, `Cancel` should be a "back" operation, whereas `Go` is a move "forward" operation.